### ISSUE TITLE
Dialog identification is failing

### ISSUE THREAD

**Issue by amay1980** [drachtio/drachtio-srf#115]
Hi,
I have the following scenario:  A communicates with B through a SIP proxy. The SIP proxy routes the call to drachtio and then drachtio routes it back to the proxy. This happens at least 3 times to apply different services and eventually the call is routed to the final destination B.
The interesting case here is that there are several B2BUA instances in same drachtio server and they are talking to each other. I made several test calls and they were failing, because a couple of the drachtio UASs failed to recognise the ACK's received. I realised that the reason for that is that drachtio was creating a wrong ACK as it was swapping the content of the From and the To headers as compared with the received 200OK (which is wrong).  It looked like drachtio was not properly recognising the dialog. In other words it was identifying the 200OK as pertaining to a  dialog at the UAS  as opposed to the dialog at the UAC.
RFC 3261, by having the dialog id be composed by callid, from tag and to tag, specifically avoids those issues. However it would seem as if drachtio was not identifying the dialogs properly.
To confirm this hypothesis , I printed the uac.id and uas.id after creation of each of the 3 B2BUA and this is what I got:

call connected
UAC: 97c8d0db-a650-123a-259f-0800272d57ee;from-tag=y4HggX7ZvcNNj
UAS: 97c07dd2-a650-123a-259f-0800272d57ee;from-tag=XUrQe2pvZ3y2p
call connected
UAC: 97c07dd2-a650-123a-259f-0800272d57ee;from-tag=XUrQe2pvZ3y2p
UAS: 97b963f9-a650-123a-259f-0800272d57ee;from-tag=vjZyc75r2t8FB
call connected
UAC: 97b963f9-a650-123a-259f-0800272d57ee;from-tag=vjZyc75r2t8FB
UAS: fac1b765-5239-4362-97a2-624ea58fbe0b;from-tag=ce00485e-6ee5-421c-a1d2-9d45989a2776

effectively, as can be seen, UAS id of first B2BUA coincides with UAS id of second dialog!! Therefore , if drachtio server is just looking at the dialog id for the 200OK in order to identify the context for sending the ACK, it might get it wrong. And this is precisely what is observed in the tests

In conclusion, I think there is an issue in how messages are mapped to dialogs, which manifests itself when one drachtio B2BUA talks to another B2BUA in the same server (through a proxy , in my case).

Thanks
